packages/bridge-ui/src/erc20/bridge.ts

```diff
@@ -0,0 +1,104 @@
+import { BigNumber, Contract, Signer } from "ethers";
+import type { Transaction } from "ethers";
+import type { ApproveOpts, Bridge, BridgeOpts } from "../domain/bridge";
+import TokenVault from "../constants/abi/TokenVault";
+import ERC20 from "../constants/abi/ERC20";
+
+class ERC20Bridge implements Bridge {
+  private async spenderRequiresAllowance(
+    tokenAddress: string,
+    signer: Signer,
+    amount: BigNumber,
+    bridgeAddress: string
+  ): Promise<boolean> {
+    const contract: Contract = new Contract(tokenAddress, ERC20, signer);
+    const allowance: BigNumber = await contract.allowance(
+      await signer.getAddress(),
```

Comment:
Minor nitpick: would be better to do this on a separate line and assign to a variable.